hikari/internal/cache.py

```diff
@@ -132,28 +145,32 @@ def __getitem__(self, key: KeyT) -> ValueT:
         return entry
 
     def __iter__(self) -> typing.Iterator[KeyT]:
-        if self._predicate is None:
-            return iter(self._data)
-        else:
-            return (key for key, value in self._data.items() if self._predicate(value))
+        return iter(self._data)
 
     def __len__(self) -> int:
-        if self._predicate is None:
-            return len(self._data)
-        else:
-            return sum(1 for value in self._data.values() if self._predicate(value))
+        return len(self._data)
 
-    def get_item_at(self, index: int) -> ValueT:
-        current_index = -1
+    @typing.overload
+    def get_item_at(self, index: int, /) -> ValueT:
+        ...
 
-        for key, value in self._data.items():
-            if self._predicate is None or self._predicate(value):
-                index += 1
+    @typing.overload
+    def get_item_at(self, index: slice, /) -> typing.Sequence[ValueT]:
+        ...
 
-            if current_index == index:
-                return self[key]
+    def get_item_at(self, index: typing.Union[slice, int]) -> typing.Union[ValueT, typing.Sequence[ValueT]]:
```

Comment:
probably want to update the typing for this method on hikari.api.cache.CacheView and the few other classes which inherit from it (plus you forgot the `, /`)